Make collector unit test tracebacks more readable

```console
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

AssertionError: 'Traceback (most recent call last):\n  File "/Users/foo/Developer/quality-time/components/collector/src/base_collectors/source_collector.py", line 155, in __safely_parse_source_responses\n    return await self._parse_source_responses(responses)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/Users/foo/Developer/quality-time/components/collector/src/base_collectors/source_collector.py", line 169, in _parse_source_responses\n    included_entities = Entities([entity for entity in entities if self._include_entity(entity)])\n                                                                   ~~~~~~~~~~~~~~~~~~~~^^^^^^^^\n  File "/Users/foo/Developer/quality-time/components/collector/src/base_collectors/source_collector.py", line 462, in _include_entity\n    and self._request_matches_draft_status(entity)\n        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^\n  File "/Users/foo/Developer/quality-time/components/collector/src/base_collectors/source_collector.py", line 489, in _request_matches_draft_status\n    ignore_drafts = self._parameter("ignore_draft_merge_requests")\n  File "/Users/foo/Developer/quality-time/components/collector/src/base_collectors/source_collector.py", line 92, in _parameter\n    return self.__parameters.get(parameter_key, quote)\n           ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^\n  File "/Users/foo/Developer/quality-time/components/collector/src/model/parameters.py", line 45, in get\n    parameter_info = DATA_MODEL.sources[self.__source["type"]].parameters[parameter_key]\n                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^\nKeyError: \'ignore_draft_merge_requests\'\n' is not None
 ```

--- a/components/collector/tests/source_collectors/source_collector_test_case.py
+++ b/components/collector/tests/source_collectors/source_collector_test_case.py
@@ -114,11 +114,17 @@ class SourceCollectorTestCase(unittest.IsolatedAsyncioTestCase):
     ) -> None:
         """Assert that the measurement has the expected attributes."""
         for attribute_key in ("connection_error", "parse_error"):
+            error_message = getattr(measurement.sources[source_index], attribute_key)
             if (attribute_value := attributes.get(attribute_key)) is not None:
                 attributes.update(value=None, total=None, entities=None)
-                self.assertIn(attribute_value, getattr(measurement.sources[source_index], attribute_key))
+                self.assertIn(attribute_value, error_message)
             else:
-                self.assertIsNone(getattr(measurement.sources[source_index], attribute_key))
+                # If the error message contains a traceback, make sure it's neatly printed:
+                if error_message and error_message.startswith("Traceback"):  # pragma: no cover
+                    error_message_to_check, error_message_to_print = error_message[:50] + "...", error_message
+                else:
+                    error_message_to_check, error_message_to_print = error_message, None
+                self.assertIsNone(error_message_to_check, error_message_to_print)
         for attribute_key in ("value", "total", "entities", "api_url", "landing_url"):
             if (attribute_value := attributes.get(attribute_key, "value not specified")) != "value not specified":
                 self.__assert_measurement_source_attribute(attribute_key, attribute_value, measurement, source_index)

